Source/Core/Core/AchievementManager.cpp

```diff
@@ -239,6 +249,38 @@ void AchievementManager::DoFrame()
   }
 }
 
+bool AchievementManager::CanPause()
+{
+  u32 frames_to_next_pause = 0;
+  bool can_pause = rc_client_can_pause(m_client, &frames_to_next_pause);
+  if (!can_pause)
+  {
+    OSD::AddMessage("Cannot spam pausing in hardcore mode.", OSD::Duration::VERY_LONG,
+                    OSD::Color::RED);
+    OSD::AddMessage(
+        fmt::format("Can pause in {} seconds.", static_cast<float>(frames_to_next_pause) / 60),
+        OSD::Duration::VERY_LONG, OSD::Color::RED);
+  }
+  return can_pause;
+}
+
+void AchievementManager::DoIdle()
+{
+  m_idle_thread = std::make_unique<std::thread>([]() {
+    while (Core::GetState(*AchievementManager::GetInstance().m_system) == Core::State::Paused &&
+           AchievementManager::GetInstance().m_background_execution_allowed)
+    {
+      Common::SleepCurrentThread(1000);
+      std::lock_guard lg{AchievementManager::GetInstance().m_lock};
+      auto* client = AchievementManager::GetInstance().m_client;
+      if (!client || !AchievementManager::GetInstance().IsGameLoaded())
+        return;
+      rc_client_idle(client);
+    }
+  });
+  m_idle_thread->detach();
```

Comment:
There's still something iffy here. If you detach the thread immediately why store it in a member variable at all?

I also generally think that casting a thread into the ether and hoping it synchronizes itself with regular polls is a poor idea (you cannot guarantee that this ever finishes, because in theory every time it polls the core state might be paused at this very moment, even if it was non-paused between the polls), but I suppose it's *okay enough* to work in practice and we can always refactor it later.